homeassistant/components/surepetcare/__init__.py

```diff
@@ -42,113 +41,63 @@
                 {
                     vol.Required(CONF_USERNAME): cv.string,
                     vol.Required(CONF_PASSWORD): cv.string,
-                    vol.Optional(CONF_FEEDERS): vol.All(
-                        cv.ensure_list, [cv.positive_int]
-                    ),
-                    vol.Optional(CONF_FLAPS): vol.All(
-                        cv.ensure_list, [cv.positive_int]
-                    ),
-                    vol.Optional(CONF_PETS): vol.All(cv.ensure_list, [cv.positive_int]),
-                    vol.Optional(CONF_SCAN_INTERVAL): cv.time_period,
-                },
-                cv.deprecated(CONF_FEEDERS),
-                cv.deprecated(CONF_FLAPS),
-                cv.deprecated(CONF_PETS),
-                cv.deprecated(CONF_SCAN_INTERVAL),
+                }
             )
         )
     },
     extra=vol.ALLOW_EXTRA,
 )
 
 
-async def async_setup(hass: HomeAssistant, config: dict) -> bool:
-    """Set up the Sure Petcare integration."""
-    conf = config[DOMAIN]
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up."""
+
     hass.data.setdefault(DOMAIN, {})
 
     try:
         surepy = Surepy(
-            conf[CONF_USERNAME],
-            conf[CONF_PASSWORD],
-            auth_token=None,
+            entry.data[CONF_USERNAME],
+            entry.data[CONF_PASSWORD],
+            auth_token=entry.data[CONF_TOKEN] if CONF_TOKEN in entry.data else None,
             api_timeout=SURE_API_TIMEOUT,
             session=async_get_clientsession(hass),
         )
     except SurePetcareAuthenticationError:
         _LOGGER.error("Unable to connect to surepetcare.io: Wrong credentials!")
         return False
```

Comment:
Should we trigger a reauth flow in this case? Probably something for a future PR